### PR TITLE
Add ALBERT masked language model

### PR DESCRIPTION
**Pull request by antimora**
Depends on https//github.com/tracel-ai/burn/pull/4410 fixes

## Summary

- Add ALBERT (A Lite BERT) masked language model implementation using the Burn deep learning framework
- Support all four v2 variants (base, large, xlarge, xxlarge) with pretrained weights loaded from HuggingFace
- Model configurations are read from HuggingFace's `config.json`, so variant-specific architectures are not hardcoded
- Inference results show high accuracy — all variants correctly predict "paris" as the top or near-top answer for the masked sentence "The capital of france is [MASK]."

## Benchmark Results

ALBERT BaseV2 inference (forward pass), 10 samples, with GPU warmup:

| Backend | Fastest | Median | Mean |
|---------|---------|--------|------|
| NdArray (CPU) | 67.54 ms | 68.55 ms | 69.32 ms |
| WGPU (GPU) | 15.80 ms | 51.68 ms | 62.71 ms |

WGPU fastest is ~4.3x faster than NdArray CPU. Run with:
```bash
cargo bench --bench inference --features "pretrained,ndarray,wgpu"
```

## Test plan

- [x] Run `cargo test -p albert-burn` to verify integration tests pass
- [x] Run inference example: `cargo run -p albert-burn --example inference --release --features pretrained`
- [x] Verify pretrained weights load correctly for each variant
- [x] Run benchmarks: `cargo bench --bench inference --features "pretrained,ndarray,wgpu"`

### CODE REVIEW

File: albert-burn/README.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+- `cuda` - CUDA backend
+- `tch-cpu` - LibTorch CPU backend
+- `tch-gpu` - LibTorch GPU backend
+
+## Example
+
+Run the fill-mask inference example:
+
+```bash
+cargo run --example inference --features ndarray --release
```
Comment (laggui) [inline albert-burn/README.md]: That doesn't work, even for other backends e.g.

```
cargo run --example inference --features cuda --release
```

yields
```
error: target `inference` in package `albert-burn` requires the features: `pretrained`, `ndarray`
Consider enabling them by passing, e.g., `--features="pretrained ndarray"`
```

**Review by laggui — CHANGES_REQUESTED** [review]
Just a couple of comments

File: albert-burn/README.md (hunk 1 above)
Comment (antimora) [reply albert-burn/README.md]: Fixed in ba475f3. Updated to use `--features "pretrained,ndarray"`.

File: albert-burn/tests/integration_test.rs [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+/// After 12 shared transformer layers, these accumulate to ~1e-4 relative error.
+const REL_TOL: f32 = 5e-4;
+
+fn rel_diff(actual: f32, expected: f32) -> f32 {
+    let abs_diff = (actual - expected).abs();
+    let scale = expected.abs().max(1.0);
+    abs_diff / scale
+}
+
+fn assert_close(actual: f32, expected: f32, label: &str) {
```
[the remark this answers is not in the document]
Comment (antimora) [reply albert-burn/tests/integration_test.rs]: Agreed. Kept the custom impl since it provides more descriptive error messages with labels for each assertion, but `TensorData::assert_approx_eq` would work too.

**Review by laggui — APPROVED** [review]

**Comment by laggui** [comment]
@antimora not sure why this broke main CI when merging

https://github.com/tracel-ai/models/actions/runs/21674714690/job/62491679306
```
   Compiling burn-store v0.20.1
error[E0308]: mismatched types
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/burn-store-0.20.1/src/pytorch/lazy_data.rs:90:25
   |
87 |             file_list.push((name, offset, compressed_size));
   |             ---------      ------------------------------- this argument has type `(std::string::String, std::option::Option<u64>, u64)`...
   |             |
   |             ... which causes `file_list` to have type `std::vec::Vec<(std::string::String, std::option::Option<u64>, u64)>`
...
90 |         Ok(Self { path, file_list })
   |                         ^^^^^^^^^ expected `Vec<(String, u64, u64)>`, found `Vec<(String, Option<u64>, u64)>`
   |
   = note: expected struct `std::vec::Vec<(std::string::String, u64, _)>`
              found struct `std::vec::Vec<(std::string::String, std::option::Option<u64>, _)>`

   Compiling burn v0.20.1
For more information about this error, try `rustc --explain E0308`.
```

/edit: ahhh shoot looks like there is a breaking change in zip 7.3.0 which was just released... even though it was a minor version. `file.data_start()` now appears to return an `Option<u64>`.
